forc/src/cli/commands/completions.rs

```diff
@@ -11,16 +46,138 @@ pub struct Command {
+                    plugins.insert(command_info.name.to_owned(), command_info);
+                }
+            }
+        }
+    });
+
+    let mut plugins = plugins.into_values().collect::<Vec<_>>();
+    cmd.subcommands.append(&mut plugins);
+
+    let mut cmd = cmd.to_clap();
+    match target {
+        Target::Bash => print_completions(clap_complete::Shell::Bash, &mut cmd, writer),
+        Target::Zsh => print_completions(clap_complete::Shell::Zsh, &mut cmd, writer),
+        Target::Fish => print_completions(clap_complete::Shell::Fish, &mut cmd, writer),
+        Target::Elvish => print_completions(clap_complete::Shell::Elvish, &mut cmd, writer),
+        Target::PowerShell => print_completions(clap_complete::Shell::PowerShell, &mut cmd, writer),
+        Target::Fig => print_completions(clap_complete_fig::Fig, &mut cmd, writer),
+    }
+}
+
+fn is_writable<P: AsRef<Path>>(path: P) -> bool {
+    if let Ok(metadata) = metadata(path) {
+        return !metadata.permissions().readonly();
+    }
+    false
 }
 
 pub(crate) fn exec(command: Command) -> ForcResult<()> {
-    let mut cmd = super::super::Opt::command();
-    print_completions(command.shell, &mut cmd);
+    let target = command.target.unwrap_or_else(|| {
+        if let Ok(shell) = std::env::var("SHELL") {
+            match Path::new(shell.as_str())
+                .file_name()
+                .unwrap()
+                .to_str()
+                .unwrap()
+            {
+                "bash" => Target::Bash,
+                "zsh" => Target::Zsh,
+                "fish" => Target::Fish,
+                "pwsh" => Target::PowerShell,
+                "elvish" => Target::Elvish,
+                _ => Target::Bash,
+            }
+        } else {
+            Target::Bash
+        }
+    });
+
+    // Where to store the autocomplete script, ZSH requires a special path, the other shells are
+    // stored in the same path and they are referenced form their respective config files
+    let dir = home::home_dir().map(|p| p.display().to_string()).unwrap();
+    let forc_autocomplete_path = match target {
+        Target::Zsh => {
+            let x = std::process::Command::new("zsh")
+                .arg("-c")
+                .arg("echo $fpath")
+                .output()
+                .expect("Cannot read $FPATH env variable")
+                .stdout;
+            let paths = String::from_utf8_lossy(&x)
+                .split(' ')
+                .filter(|path| is_writable(Path::new(path)))
+                .map(|x| x.to_owned())
+                .collect::<Vec<_>>();
+            format!(
+                "{}/_forc",
+                paths.first().expect("No writable path found for zsh")
+            )
+        }
+        _ => format!("{}/.forc.autocomplete", dir),
+    };
+
+    let mut file = File::create(&forc_autocomplete_path).unwrap_or_else(|_| {
+        panic!("Cannot write to the autocomplete file in path {forc_autocomplete_path}")
+    });
```

Comment:
We should definitely ask users before writing to their shell configuration. We should also tell the user what files are being written to.